src/app/utils.js

```diff
@@ -230,7 +230,22 @@ export const debounce = (fn, wait) => {
   };
 };
 
+/**
+  Returns a bool on whether a user can admin controllers
+  @param {Obj} conn The user connection.
+  @returns {Bool} True or false
+*/
 export const userIsControllerAdmin = (conn) => {
   const controllerAccess = conn?.info?.user?.controllerAccess;
   return ["superuser", "admin"].includes(controllerAccess);
 };
+
+/**
+  Returns the charm name for provided namespace.
+  @param {String} namespace The fully qualified charm name.
+  @returns {String} The charm name.
+*/
+export const extractCharmName = (namespace) => {
+  if (!namespace) return null;
```

Comment:
Because the 'happy' path returns a `string` type, the failure path in this case should probably as well.